### PR TITLE
Added the connection poolers blog post

### PR DESCRIPTION
**Pull request by rzavalet**
The corresponding google docs is here: https://docs.google.com/document/d/1Csz0khn31gZQ17w-Ws3X-n0Mc3IK_SC_DD61qWN-Ugk/edit

### CODE REVIEW

File: landing/src/content/blog/2024-02-13-connection-poolers/index.md [hunk 1]
```diff
@@ -0,0 +1,203 @@
+---
+slug: postgres-connection-poolers
+title: 'Detailed analysis of PostgreSQL connection poolers'
+authors: [rjzv]
+tags: [postgres, "connection pool", pgbouncer, pgcat, supavisor, pgbench]
```
Comment (rzavalet) [inline landing/src/content/blog/2024-02-13-connection-poolers/index.md:5]: Missing a thumbnail image.

File: landing/src/content/blog/2024-02-13-connection-poolers/index.md [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+---
+
+Creating a connection to your Postgres database to execute a short-lived query is expensive. Several people have measured the overhead of Postgres connections and some locate them in the range of [1.3MB of memory per connection](https://stackoverflow.blog/2020/10/14/improve-database-performance-with-connection-pooling/) and others in the range of [2MB](https://blog.anarazel.de/2020/10/07/measuring-the-memory-overhead-of-a-postgres-connection/). In addition, there is also the overhead of having to [fork a new process](https://www.postgresql.org/docs/current/connect-estab.html) in the database server. 
+
+To alleviate these problems, typically people use a connection pooler. However, for Postgres, there are several options available. Some popular ones are Pgbouncer, Pgcat, Odyssey, PgAgroal, PgPool and Supavisor.
```
Comment (samay-sharma) [inline landing/src/content/blog/2024-02-13-connection-poolers/index.md:10]: link to these product pages

File: landing/src/content/blog/2024-02-13-connection-poolers/index.md [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+### Throughput and Latency for large connection count
+
+Let’s see what happens when we stress the connection poolers using more clients. The following two graphs show my results when the number of concurrent clients varied in the range of 250-2500:
+
+![Average Latency for large connection count](./006_average_latency_many_conns.png)
+
+![Average Throughput for large connection count](./007_average_throughput_many_conns.png)
+
+For the three poolers, the latency of executing a simple query continues growing. This is because, beyond a certain point, the CPU becomes a bottleneck, and connection requests begin to pile up. I suspect this could be mitigated with a more powerful machine.
```
Comment (samay-sharma) [inline landing/src/content/blog/2024-02-13-connection-poolers/index.md:115]: For the three poolers, the latency of executing a simple query continues **to grow**

File: landing/src/content/blog/2024-02-13-connection-poolers/index.md [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+
+We can see that PgCat performs better (higher throughput, lower latency), reaching 59K tps. In contrast, PgBouncer peaks at 44,096 tps and degrades to a steady state of 25,000-30,000 tps beyond 75 concurrent connections. Supavisor peaks at about  21,700 tps, but remains in steady state.
```
Comment (samay-sharma) [inline landing/src/content/blog/2024-02-13-connection-poolers/index.md:117]: Let's bold the numbers

File: landing/src/content/blog/2024-02-13-connection-poolers/index.md [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+| **Repo Popularity (Stars in GH)**                                  | 2.5K                                    | 2.3K                                 | 1.4K                                   |
+| **Language of implementation**                                     | C                                       | Rust                                 | Elixir                                 |
+| **Installation/Setup Complexity**                                  | Easy                                    | Easy                                 | Medium, several moving parts.          |
+| **Multi-threaded**                                                 | No                                      | Yes                                  | Yes                                    |
+| **Application Changes**                                            | Connection string                       | Connection string                    | Connection string                      |
+| **Experiments with E2-standard-8 (8 vCPUs, 4 cores, 32GB memory)** |                                         |                                      |                                        |
```
Comment (samay-sharma) [inline landing/src/content/blog/2024-02-13-connection-poolers/index.md:182]: This doesn't render well. Can you break it into two tables?

File: landing/src/content/blog/2024-02-13-connection-poolers/index.md [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+| **Max Throughput**                                                 | 44,096 tps @ 50 clients                 | 59,051 tps @ 1,250 clients           | 21,708 tps @ 100 clients               |
+| **Latency @ Max throughput**                                       | 1.13 ms                                 | 64.3 ms                              | 4.6 ms                                 |
+| **Latency at 1,250 concurrent connections**                         | 47.2 ms                                 | 21.1 ms                              | 64.37 ms                               |
+
+In this post, we compared connection poolers for Postgres across different axes.
+
+Of the three, Supavisor is the one that requires more steps to begin using it. However, the way it handles tenants is convenient for modern cloud environments. Other than that, once the poolers are adequately set up, the only change required in your application is the connection string.
+
+From my experiments, I found that PgCat performs better as it delivers higher throughput (more than 2X compared to PgBouncer and Supavisor) while supporting more concurrent clients.
```
Comment (samay-sharma) [inline landing/src/content/blog/2024-02-13-connection-poolers/index.md:192]: The 2x number is not accurate all the time. Can you qualify when it displays this throughput?

File: landing/src/content/blog/2024-02-13-connection-poolers/index.md (hunk 3 above)
Comment (rzavalet) [reply landing/src/content/blog/2024-02-13-connection-poolers/index.md]: Fixed

File: landing/src/content/blog/2024-02-13-connection-poolers/index.md [hunk 7]
```diff
@@ -0,0 +1,206 @@
+---
+slug: postgres-connection-poolers
+title: 'Detailed analysis of PostgreSQL connection poolers'
```
Comment (rzavalet) [inline landing/src/content/blog/2024-02-13-connection-poolers/index.md:3]: I asked chatgpt to propose alternative titles. Here are a few that caught my attention:

"Exploring PostgreSQL Connection Poolers: A Comprehensive Comparison"
"Navigating PostgreSQL Connection Poolers: An In-Depth Evaluation"
"Diving Deep into PostgreSQL Connection Poolers: A Comparative Study"
"Deciphering PostgreSQL Connection Poolers: A Comprehensive Breakdown"
"Comparative Analysis of PostgreSQL Connection Poolers: Unveiling Performance"
"Understanding PostgreSQL Connection Poolers: Analyzing Performance and Features"


For our current title, maybe something like this sounds a bit more catchy :thinking: :

"Exploring PostgreSQL connection poolers: A detailed analysis".

Comment (rzavalet) [reply landing/src/content/blog/2024-02-13-connection-poolers/index.md:3]: Actually, I think I like the  current one.

**Review by samay-sharma — APPROVED** [review]

**Pull request merged**